sherlock_test.py

```diff
@@ -0,0 +1,151 @@
+
+    Keyword Arguments:
+    username_exist_list    -- List of strings indicating usernames that should
+                              exist in all of the specified sites.
+    username_unknown_list  -- List of strings indicating usernames that should
+                              not exist in all of the specified sites.
+    site_data_all          -- Dictionary containing all of the site data.
+    verbose                -- Boolean indicating whether to give verbose output.
+    tor                    -- Boolean indicating whether to use a tor circuit for the requests.
+    unique_tor             -- Boolean indicating whether to use a new tor circuit for each request.
+
+    Return Value:
+    Boolean indicating if tests succeeded.
+    """
+
+    tests_succeeded = True
+
+    #Run test to ensure that usernames which exist are detected to exist.
+    for username in username_exist_list:
+        results = sherlock.sherlock(username,
+                                    site_data_all,
+                                    verbose=verbose,
+                                    tor=tor,
+                                    unique_tor=unique_tor
+                                   )
+        for social_network, result in results.items():
+            if result['exists'] == "yes":
+                 logging.info(f"Username '{username}' detected "
+                              f"on {social_network}."
+                             )
+            else:
+                 #At least one test failed.
+                 tests_succeeded = False
+                 logging.error(f"Username '{username}' not detected "
+                               f"on {social_network}."
+                              )
+
+    #Run test to ensure that usernames which do not exist are detected to
+    #not exist.
+    for username in username_unknown_list:
+        results = sherlock.sherlock(username,
+                                    site_data_all,
+                                    verbose=verbose,
+                                    tor=tor,
+                                    unique_tor=unique_tor
+                                   )
+        for social_network, result in results.items():
+            if result['exists'] == "no":
+                 logging.info(f"Username '{username}' not detected "
+                              f"on {social_network}."
+                             )
+            else:
+                 #At least one test failed.
+                 tests_succeeded = False
+                 logging.error(f"Username '{username}' detected "
+                               f"on {social_network}."
+                              )
+
+    return tests_succeeded
+
+
+def main():
+    # Colorama module's initialization.
+    init(autoreset=True)
+
+    version_string = f"%(prog)s {__version__}\n" +  \
+                     f"{requests.__description__}:  {requests.__version__}\n" + \
+                     f"Python:  {platform.python_version()}"
```

Comment:
Can we use () to avoid the use of \\ as discussed in PEP8?
```python
version_string = (f"%(prog)s {__version__}\n"
                      f"{requests.__description__}:  {requests.__version__}\n"
                      f"Python:  {platform.python_version()}")
```